src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -59,11 +60,33 @@ internal TableServiceClient TableClient
         {
             get
             {
-                if (!_environment.IsPlaceholderModeEnabled() && _tableClient == null && !_azureTableStorageProvider.TryCreateHostingTableServiceClient(out _tableClient))
+                if (!_environment.IsPlaceholderModeEnabled() && _tableClient == null)
                 {
-                    _logger.LogWarning("An error occurred initializing the Table Storage Client. We are unable to record diagnostic events, so the diagnostic logging service is being stopped.");
-                    _isEnabled = false;
-                    StopTimer();
+                    if (!_azureTableStorageProvider.TryCreateHostingTableServiceClient(out _tableClient))
+                    {
+                        DisableService();
+                        Logger.ServiceDisabledFailedToCreateClient(_logger);
+                        return _tableClient;
+                    }
+
+                    try
+                    {
+                        // When using RBAC, we need "Storage Table Data Contributor" as we require to list, create and delete tables and query/insert/delete entities.
+                        // Testing permissions by listing tables and creating a test table.
+                        var testTable = _tableClient.GetTableClient($"{TableNamePrefix}Check");
+                        _ = TableStorageHelpers.TableExist(testTable, _tableClient);
+                        _ = testTable.CreateIfNotExists();
```

Comment:
`testTable.CreateIfNotExists();` inherently requires both read and write permissions. Do we need to call `TableExist(...)`?

Reply:
`CreateIfNotExists` internally tries to create the table and checks if the response is a 409 (conflict) and the error code is distinct than `TableBeingDeleted`. That way it knows the table already exists. You could create tables without having read permissions on the table service (custom role for example). Since we need to query the list of tables (read permissions) we test that with `TableStorageHelpers.TableExist`.